xmake/rules/qt/qmltyperegistrar/xmake.lua

```diff
@@ -126,7 +126,8 @@ rule("qt.qmltyperegistrar")
         batchcmds:show_progress(opt.progress, "${color.build.object}compiling.qt.qmltyperegistrar %s", path.filename(sourcefile))
         batchcmds:compile(sourcefile, objectfile)
 
-        batchcmds:add_depfiles(sourcefile)
+        batchcmds:add_depvalues({importname, majorversion, minorversion})
+        batchcmds:add_depfiles(table.join(sourcebatch.sourcefiles, sourcefile))
```

Comment:
@waruqi Not sure if you're aware, but GitHub has this quite handy feature called suggestions (there is a button on the toolbar). JFYI.
```suggestion
        batchcmds:add_depvalues(importname, majorversion, minorversion)
        batchcmds:add_depfiles(sourcefile, sourcebatch.sourcefiles)
```